test/framework/e2e_framework.go

```diff
@@ -446,17 +473,25 @@ func (f *E2eFramework) DeployK8sClientConfigs(namespace string) error {
 		return err
 	}
 
+	destCfg := fmt.Sprintf(*configFileTemplate, 0)
+	kubeconfigs := filepath.Join(baseDir, *configDir)
+	inClusterKubeconfigs := filepath.Join(baseDir, *inClusterConfigDir)
 	i := 0
-	for k8sContext, _ := range f.remoteClients {
+	for k8sContext := range f.remoteClients {
 		f.logger.Info("Creating ClientConfig", "cluster", k8sContext)
-		srcCfg := fmt.Sprintf("k8ssandra-%d.yaml", i)
-		cmd := exec.Command(
-			filepath.Join("scripts", "create-clientconfig.sh"),
-			"--src-kubeconfig", filepath.Join("build", "kubeconfigs", srcCfg),
-			"--dest-kubeconfig", filepath.Join("build", "kubeconfigs", "k8ssandra-0.yaml"),
-			"--in-cluster-kubeconfig", filepath.Join("build", "kubeconfigs", "updated", srcCfg),
-			"--output-dir", filepath.Join("build", "clientconfigs", srcCfg),
-			"--namespace", namespace)
+		srcCfg := fmt.Sprintf(*configFileTemplate, i)
```

Comment:
This won't work if I want to use `~/.kube/config` to run tests against some GKE clusters for example. I recently created some GKE clusters and configured the context names to be control-plane, east, west, and central. Here is an example cmd line I used to generate a ClientConfig for the central cluster:

```
./scripts/create-clientconfig.sh --namespace k8ssandra-operator \
                                 --src-context central  \
                                 --dest-context control-plane \
                                 --output-dir clientconfigs
```
`DeployK8sClientConfigs` most other setup code assumes a directory structure and configs as created by the `setup-kind-multicluster.sh` script.

I authenticate to my GKE clusters with the `gcloud container clusters get-credentials` command. It adds a context (and other stuff) in `~/.kube/config`. Tools for other cloud providers will also update the default kubeconfig. It makes sense then to also be able to use it for e2e tests. Since there may be any number of other clusters/contexts in the default config, we should require the control plane and data plane contexts to be explicitly set.

Reply:
I am sorry, but I am failing to understand how you would like this to be. In fact, I'm failing to understand when/why the in-cluster config is required, and how tests are going to be executed against a cloud provider like GKE. I would suggest that we leave this part out of this PR, then someone else more skilled than me could tackle that part later.

Reply:
When you create a kind cluster it updates `~/.kube/config` with an entry like this:

```yaml
- cluster:
    certificate-authority-data: ...
    server: https://127.0.0.1:55055
  name: kind-k8ssandra-0
```
We don't want a localhost address for the api server, so we generate a kubeconfig using the in-cluster IP of the api server. This doesn't need to be done when using real clusters.

I'm fine though with addressing this in a follow up PR. It makes the work more incremental and PRs smaller which I think is a good thing.